src/Components/Components/test/Routing/RouterTest.cs

```diff
@@ -117,34 +84,12 @@ async Task OnNavigateAsync(NavigationContext args)
             Assert.Equal(2, onNavigateInvoked);
         }
 
-        [Fact]
-        public async Task CanHandleSingleCancelledOnNavigateAsync()
-        {
-            // Arrange
-            async Task OnNavigateAsync(NavigationContext args)
-            {
-                var tcs = new TaskCompletionSource<int>();
-                tcs.TrySetCanceled();
-                await tcs.Task;
-            }
-            _renderer.OnUpdateDisplay = (renderBatch) => Assert.True(false, "OnUpdateDisplay called more than once.");
-            _router.OnNavigateAsync = OnNavigateAsync;
-
-            // Act
-            await _renderer.Dispatcher.InvokeAsync(() => _router.RunOnNavigateWithRefreshAsync("http://example.com/jan", false));
-
-            // Assert
-            Assert.Single(_renderer.HandledExceptions);
-            var unhandledException = _renderer.HandledExceptions[0];
-            Assert.Equal("OnNavigateAsync can only be cancelled via NavigateContext.CancellationToken.", unhandledException.Message);
-        }
-
```

Comment:
Instead of removing this test, can we capture the new behavior? (Canceled OnNavigate invocations will result in a new render)

Reply:
Sure.